File: GYM Questions/Recursion/Find Digit Sum.java
import java.util.*;

// https://course.acciojob.com/idle?question=ff2bc0d6-e2d3-40ae-99b0-a585940c872b

public class Main {
	static int sumOfDigi(int n) {
        if(n == 0){
			return 0;
		}

		return n%10 + sumOfDigi(n/=10);
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n;
        n = sc.nextInt();
        int result = sumOfDigi(n);
        System.out.println(result);
        sc.close();
    }
}

File: Javascript/Find Digit Sum.js
// https://course.acciojob.com/idle?question=ea62270d-e330-4190-9d03-e25e6c84d41e

const readline = require('readline').createInterface({
  input: process.stdin,
  output: process.stdout
});

function sumOfDigi(n) {
  let result = 0;
    while(n > 0){
        result += n%10;
        n = Math.floor(n/10);
    }
    return result;
}

readline.question('', n => {
  let result = sumOfDigi(parseInt(n));
  console.log(result);
  readline.close();
});
